alert-common/src/main/java/com/synopsys/integration/alert/common/SetMap.java

```diff
@@ -0,0 +1,54 @@
+package com.synopsys.integration.alert.common;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class SetMap<K, S> {
+    private final Map<K, Set<S>> map;
+
+    public SetMap(final Map<K, Set<S>> map) {
+        this.map = map;
+    }
+
+    public Set<S> getValue(K key) {
+        return map.get(key);
+    }
+
+    public Set<S> add(K key, S value) {
+        if (map.containsKey(key)) {
```

Comment:
Use computeIfAbsent here and then you can just call set.add after the call to computeIfAbsent.  It would collapse the code to a couple of lines.